mitodl/mit-open
# Add profile option for silky to settings

### What are the relevant tickets?




N/A

### Description (What does it do?)

This adds an environment variable for being able to profile API requests locally. I didn't add ti to `app.json` because we probably shouldn't be using it there (at least not without some testing).

### How can this be tested?

Set `SILKY_PYTHON_PROFILER=True` in your `env/backend.local.env`and verify that when you now run API requests that silky records cProfile results (http://api.open.odl.local:8063/silk locally, see screenshot):

![Screenshot 2024-07-15 at 14-58-19 Silky - CProfile - _api_v0_news_events_](https://github.com/user-attachments/assets/2210cb96-e659-4744-842a-de28b64460a3)
